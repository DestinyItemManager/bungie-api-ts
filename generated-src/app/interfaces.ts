/**
 * Bungie.Net API
 * These endpoints constitute the functionality exposed by Bungie.net, both for more traditional website functionality and for connectivity to Bungie video games and their related functionality.
 *
 * OpenAPI spec version: 2.9.2
 * Contact: support@bungie.com
 *
 * NOTE: This class is auto generated by the bungie-api-ts code generator program.
 * https://github.com/DestinyItemManager/bungie-api-ts
 * Do not edit these files manually.
 */

import {
  UserInfoCard
} from '../user/interfaces.js';

export interface ApiUsage {
  /** Counts for on API calls made for the time range. */
  readonly apiCalls: Series[];
  /**
   * Instances of blocked requests or requests that crossed the warn threshold during
   * the time range.
   */
  readonly throttledRequests: Series[];
}

export interface Series {
  /** Collection of samples with time and value. */
  readonly datapoints: Datapoint[];
  /** Target to which to datapoints apply. */
  readonly target: string;
}

export interface Datapoint {
  /** Timestamp for the related count. */
  readonly time: string;
  /** Count associated with timestamp */
  readonly count?: number;
}

export interface Application {
  /** Unique ID assigned to the application */
  readonly applicationId: number;
  /** Name of the application */
  readonly name: string;
  /** URL used to pass the user's authorization code to the application */
  readonly redirectUrl: string;
  /** Link to website for the application where a user can learn more about the app. */
  readonly link: string;
  /** Permissions the application needs to work */
  readonly scope: string;
  /** Value of the Origin header sent in requests generated by this application. */
  readonly origin: string;
  /** Current status of the application. */
  readonly status: ApplicationStatus;
  /** Date the application was first added to our database. */
  readonly creationDate: string;
  /** Date the application status last changed. */
  readonly statusChanged: string;
  /** Date the first time the application status entered the 'Public' status. */
  readonly firstPublished: string;
  /**
   * List of team members who manage this application on Bungie.net. Will always
   * consist of at least the application owner.
   */
  readonly team: ApplicationDeveloper[];
}

export const enum ApplicationStatus {
  /** No value assigned */
  None = 0,
  /**
   * Application exists and works but will not appear in any public catalog. New
   * applications start in this state, test applications will remain in this state.
   */
  Private = 1,
  /** Active applications that can appear in an catalog. */
  Public = 2,
  /**
   * Application disabled by the owner. All authorizations will be treated as
   * terminated while in this state. Owner can move back to private or public state.
   */
  Disabled = 3,
  /**
   * Application has been blocked by Bungie. It cannot be transitioned out of this
   * state by the owner. Authorizations are terminated when an application is in this
   * state.
   */
  Blocked = 4
}

export interface ApplicationDeveloper {
  readonly role: DeveloperRole;
  readonly apiEulaVersion: number;
  readonly user: UserInfoCard;
}

export const enum DeveloperRole {
  None = 0,
  Owner = 1,
  TeamMember = 2
}
