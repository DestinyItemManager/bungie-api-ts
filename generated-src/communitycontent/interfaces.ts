/**
 * Bungie.Net API
 * These endpoints constitute the functionality exposed by Bungie.net, both for more traditional website functionality and for connectivity to Bungie video games and their related functionality.
 *
 * OpenAPI spec version: 2.3.5
 * Contact: support@bungie.com
 *
 * NOTE: This class is auto generated by the bungie-api-ts code generator program.
 * https://github.com/DestinyItemManager/bugie-api-ts
 * Do not edit these files manually.
 */

import {
  PagedQuery
} from '../common';
import {
  PartnershipType
} from '../platform';
import {
  UserInfoCard
} from '../user/interfaces';

export const enum CommunityStatusSort {
  Viewers = 0,
  Trending = 1,
  OverallViewers = 2,
  Followers = 3
}

export interface SearchResultOfCommunityLiveStatus {
  readonly results: CommunityLiveStatus[];
  readonly totalResults: number;
  readonly hasMore: boolean;
  readonly query: PagedQuery;
  readonly replacementContinuationToken: string;
  /**
   * If useTotalResults is true, then totalResults represents an accurate count.
   *
   * If False, it does not, and may be estimated/only the size of the current page.
   *
   * Either way, you should probably always only trust hasMore.
   *
   * This is a long-held historical throwback to when we used to do paging with known
   * total results. Those queries toasted our database, and we were left to hastily
   * alter our endpoints and create backward- compatible shims, of which
   * useTotalResults is one.
   */
  readonly useTotalResults: boolean;
}

export interface CommunityLiveStatus {
  readonly dateStatusUpdated: string;
  readonly url: string;
  readonly partnershipIdentifier: string;
  readonly partnershipType: PartnershipType;
  readonly thumbnail: string;
  readonly thumbnailSmall: string;
  readonly thumbnailLarge: string;
  readonly destinyCharacterId: string;
  readonly userInfo: UserInfoCard;
  readonly currentActivityHash: number;
  readonly dateLastPlayed: string;
  readonly dateStreamStarted: string;
  readonly locale: string;
  readonly currentViewers: number;
  readonly followers: number;
  readonly overallViewers: number;
  readonly isFeatured: boolean;
  readonly title: string;
  readonly activityModeHash: number;
  readonly dateFeatured?: string;
  readonly trendingValue: number;
  readonly isSubscribable: boolean;
}
