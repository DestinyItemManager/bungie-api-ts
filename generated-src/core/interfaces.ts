/**
 * Bungie.Net API
 * These endpoints constitute the functionality exposed by Bungie.net, both for more traditional website functionality and for connectivity to Bungie video games and their related functionality.
 *
 * OpenAPI spec version: 2.9.2
 * Contact: support@bungie.com
 *
 * NOTE: This class is auto generated by the bungie-api-ts code generator program.
 * https://github.com/DestinyItemManager/bungie-api-ts
 * Do not edit these files manually.
 */

import {
  EmailSettings
} from '../user/interfaces.js';

export interface CoreSettingsConfiguration {
  readonly environment: string;
  readonly systems: { [key: string]: CoreSystem };
  readonly ignoreReasons: CoreSetting[];
  readonly forumCategories: CoreSetting[];
  readonly groupAvatars: CoreSetting[];
  readonly destinyMembershipTypes: CoreSetting[];
  readonly recruitmentPlatformTags: CoreSetting[];
  readonly recruitmentMiscTags: CoreSetting[];
  readonly recruitmentActivities: CoreSetting[];
  readonly userContentLocales: CoreSetting[];
  readonly systemContentLocales: CoreSetting[];
  readonly clanBannerDecals: CoreSetting[];
  readonly clanBannerDecalColors: CoreSetting[];
  readonly clanBannerGonfalons: CoreSetting[];
  readonly clanBannerGonfalonColors: CoreSetting[];
  readonly clanBannerGonfalonDetails: CoreSetting[];
  readonly clanBannerGonfalonDetailColors: CoreSetting[];
  readonly clanBannerStandards: CoreSetting[];
  readonly destiny2CoreSettings: Destiny2CoreSettings;
  readonly emailSettings: EmailSettings;
  readonly fireteamActivities: CoreSetting[];
}

export interface CoreSystem {
  readonly enabled: boolean;
  readonly parameters: { [key: string]: string };
}

export interface CoreSetting {
  readonly identifier: string;
  readonly isDefault: boolean;
  readonly displayName: string;
  readonly summary: string;
  readonly imagePath: string;
  readonly childSettings: CoreSetting[];
}

export interface Destiny2CoreSettings {
  readonly collectionRootNode: number;
  readonly badgesRootNode: number;
  readonly recordsRootNode: number;
  readonly medalsRootNode: number;
  readonly metricsRootNode: number;
  readonly activeTriumphsRootNodeHash: number;
  readonly activeSealsRootNodeHash: number;
  readonly legacyTriumphsRootNodeHash: number;
  readonly legacySealsRootNodeHash: number;
  readonly medalsRootNodeHash: number;
  readonly exoticCatalystsRootNodeHash: number;
  readonly loreRootNodeHash: number;
  readonly currentRankProgressionHashes: number[];
  readonly undiscoveredCollectibleImage: string;
  readonly ammoTypeHeavyIcon: string;
  readonly ammoTypeSpecialIcon: string;
  readonly ammoTypePrimaryIcon: string;
  readonly currentSeasonalArtifactHash: number;
  readonly currentSeasonHash?: number;
  readonly futureSeasonHashes: number[];
  readonly pastSeasonHashes: number[];
}

export interface GlobalAlert {
  readonly AlertKey: string;
  readonly AlertHtml: string;
  readonly AlertTimestamp: string;
  readonly AlertLink: string;
  readonly AlertLevel: GlobalAlertLevel;
  readonly AlertType: GlobalAlertType;
  readonly StreamInfo: StreamInfo;
}

export const enum GlobalAlertLevel {
  Unknown = 0,
  Blue = 1,
  Yellow = 2,
  Red = 3
}

export const enum GlobalAlertType {
  GlobalAlert = 0,
  StreamingAlert = 1
}

export interface StreamInfo {
  readonly ChannelName: string;
}
