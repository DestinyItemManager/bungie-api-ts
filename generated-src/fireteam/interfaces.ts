/**
 * Bungie.Net API
 * These endpoints constitute the functionality exposed by Bungie.net, both for more traditional website functionality and for connectivity to Bungie video games and their related functionality.
 *
 * OpenAPI spec version: 2.3.5
 * Contact: support@bungie.com
 *
 * NOTE: This class is auto generated by the bungie-api-ts code generator program.
 * https://github.com/DestinyItemManager/bugie-api-ts
 * Do not edit these files manually.
 */

import {
  PagedQuery
} from '../common';
import {
  UserInfoCard
} from '../user/interfaces';

export const enum FireteamActivityType {
  All = 0,
  Raid = 1,
  Crucible = 2,
  Trials = 3,
  Nightfall = 4,
  Anything = 5,
  Gambit = 6,
  BlindWell = 7,
  EscalationProtocol = 8,
  Forge = 9,
  Reckoning = 10
}

export const enum FireteamDateRange {
  All = 0,
  Now = 1,
  TwentyFourHours = 2,
  FortyEightHours = 3,
  ThisWeek = 4
}

export const enum FireteamPlatform {
  Unknown = 0,
  Playstation4 = 1,
  XboxOne = 2,
  Blizzard = 3
}

export const enum FireteamPublicSearchOption {
  PublicAndPrivate = 0,
  PublicOnly = 1,
  PrivateOnly = 2
}

export const enum FireteamSlotSearch {
  NoSlotRestriction = 0,
  HasOpenPlayerSlots = 1,
  HasOpenPlayerOrAltSlots = 2
}

export interface SearchResultOfFireteamSummary {
  readonly results: FireteamSummary[];
  readonly totalResults: number;
  readonly hasMore: boolean;
  readonly query: PagedQuery;
  readonly replacementContinuationToken: string;
  /**
   * If useTotalResults is true, then totalResults represents an accurate count.
   *
   * If False, it does not, and may be estimated/only the size of the current page.
   *
   * Either way, you should probably always only trust hasMore.
   *
   * This is a long-held historical throwback to when we used to do paging with known
   * total results. Those queries toasted our database, and we were left to hastily
   * alter our endpoints and create backward- compatible shims, of which
   * useTotalResults is one.
   */
  readonly useTotalResults: boolean;
}

export interface FireteamSummary {
  readonly fireteamId: string;
  readonly groupId: string;
  readonly platform: FireteamPlatform;
  readonly activityType: FireteamActivityType;
  readonly isImmediate: boolean;
  readonly scheduledTime?: string;
  readonly ownerMembershipId: string;
  readonly playerSlotCount: number;
  readonly alternateSlotCount?: number;
  readonly availablePlayerSlotCount: number;
  readonly availableAlternateSlotCount: number;
  readonly title: string;
  readonly dateCreated: string;
  readonly dateModified?: string;
  readonly isPublic: boolean;
  readonly locale: string;
  readonly isValid: boolean;
  readonly datePlayerModified: string;
}

export interface SearchResultOfFireteamResponse {
  readonly results: FireteamResponse[];
  readonly totalResults: number;
  readonly hasMore: boolean;
  readonly query: PagedQuery;
  readonly replacementContinuationToken: string;
  /**
   * If useTotalResults is true, then totalResults represents an accurate count.
   *
   * If False, it does not, and may be estimated/only the size of the current page.
   *
   * Either way, you should probably always only trust hasMore.
   *
   * This is a long-held historical throwback to when we used to do paging with known
   * total results. Those queries toasted our database, and we were left to hastily
   * alter our endpoints and create backward- compatible shims, of which
   * useTotalResults is one.
   */
  readonly useTotalResults: boolean;
}

export interface FireteamResponse {
  readonly Summary: FireteamSummary;
  readonly Members: FireteamMember[];
  readonly Alternates: FireteamMember[];
}

export interface FireteamMember {
  readonly destinyUserInfo: UserInfoCard;
  readonly bungieNetUserInfo: UserInfoCard;
  readonly characterId: string;
  readonly dateJoined: string;
  readonly hasMicrophone: boolean;
  readonly lastPlatformInviteAttemptDate: string;
  readonly lastPlatformInviteAttemptResult: FireteamPlatformInviteResult;
}

export const enum FireteamPlatformInviteResult {
  None = 0,
  Success = 1,
  AlreadyInFireteam = 2,
  Throttled = 3,
  ServiceError = 4
}
