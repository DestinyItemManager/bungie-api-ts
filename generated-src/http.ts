export type HttpQueryParams = Record<string, string>;

export interface HttpClientConfig {
  method: 'GET' | 'POST';
  url: string;
  /** Query string params */
  params?: HttpQueryParams;
  /** Request body */
  body?: any;
}

/**
 * A generic, minimal type for providing an HTTP client function.
 * This gets the necessary config provided as an argument, and then
 * should produce a promise for the parsed JSON as a result. The API
 * helper functions will use this to return the right type.
 */
export type HttpClient = <Return>(config: HttpClientConfig) => Promise<Return>;

export function get<Return>(
  http: HttpClient,
  url: string,
  params?: HttpQueryParams
): Promise<Return> {
  return http(
    params
      ? {
          method: 'GET',
          url,
          params,
        }
      : {
          method: 'GET',
          url,
        }
  );
}

export function post<Return>(http: HttpClient, url: string, body?: any): Promise<Return> {
  return http(
    body
      ? {
          method: 'POST',
          url,
          body,
        }
      : {
          method: 'POST',
          url,
        }
  );
}
