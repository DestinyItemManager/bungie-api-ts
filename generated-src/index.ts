/**
 * Bungie.Net API
 * These endpoints constitute the functionality exposed by Bungie.net, both for more traditional website functionality and for connectivity to Bungie video games and their related functionality.
 *
 * OpenAPI spec version: 2.9.2
 * Contact: support@bungie.com
 *
 * NOTE: This class is auto generated by the bungie-api-ts code generator program.
 * https://github.com/DestinyItemManager/bungie-api-ts
 * Do not edit these files manually.
 */

import * as AppImport from './app/index.js';
import * as UserImport from './user/index.js';
import * as ContentImport from './content/index.js';
import * as ForumImport from './forum/index.js';
import * as GroupV2Import from './groupv2/index.js';
import * as TokensImport from './tokens/index.js';
import * as Destiny2Import from './destiny2/index.js';
import * as CommunityContentImport from './communitycontent/index.js';
import * as TrendingImport from './trending/index.js';
import * as FireteamImport from './fireteam/index.js';
import * as CoreImport from './core/index.js';

export const App = AppImport;
export const User = UserImport;
export const Content = ContentImport;
export const Forum = ForumImport;
export const GroupV2 = GroupV2Import;
export const Tokens = TokensImport;
export const Destiny2 = Destiny2Import;
export const CommunityContent = CommunityContentImport;
export const Trending = TrendingImport;
export const Fireteam = FireteamImport;
export const Core = CoreImport;
