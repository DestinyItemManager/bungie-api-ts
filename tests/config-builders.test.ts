import {
  getDestinyManifest,
  getDestinyEntityDefinition,
  searchDestinyPlayer,
  getLinkedProfiles,
  getProfile,
  getCharacter,
  getClanWeeklyRewardState,
  getItem,
  getVendors,
  getVendor,
  getPublicVendors,
  getCollectibleNodeDetails,
  transferItem,
  pullFromPostmaster,
  equipItem,
  equipItems,
  setItemLockState,
  setQuestTrackedState,
  insertSocketPlug,
  getPostGameCarnageReport,
  reportOffensivePostGameCarnageReportPlayer,
  getHistoricalStatsDefinition,
  getClanLeaderboards,
  getClanAggregateStats,
  getLeaderboards,
  getLeaderboardsForCharacter,
  searchDestinyEntities,
  getHistoricalStats,
  getHistoricalStatsForAccount,
  getActivityHistory,
  getUniqueWeaponHistory,
  getDestinyAggregateActivityStats,
  getPublicMilestoneContent,
  getPublicMilestones,
  awaInitializeRequest,
  awaProvideAuthorizationResult,
  awaGetActionToken,
} from '../generated-src/destiny2/api';
import { getDestinyManifestSlice } from '../generated-src/destiny2/manifest';
import { HttpClient, HttpClientConfig } from '../generated-src/http';
import { DestinyComponentType } from '../generated-src/destiny2/interfaces';
import fetch from 'node-fetch';

// instead of actually performing a fetch,
// this fake HttpClient returns the HttpConfig it would have used to make the fetch
// so we can check of the HttpConfig was built right
const h = ((httpConfig?: any) => Promise.resolve(httpConfig)) as HttpClient;

// sample values for the "body" argument
const b = {} as any;

// sample values for parameters
const p = {
  characterId: '1234658790',
  components: [
    DestinyComponentType.Vendors,
    DestinyComponentType.VendorSales,
    DestinyComponentType.ItemInstances,
    DestinyComponentType.ItemObjectives,
    DestinyComponentType.ItemStats,
    DestinyComponentType.ItemSockets,
    DestinyComponentType.ItemTalentGrids,
    DestinyComponentType.ItemCommonData,
    DestinyComponentType.CurrencyLookups,
    DestinyComponentType.ItemPlugStates,
    DestinyComponentType.ItemReusablePlugs,
    DestinyComponentType.ItemPlugObjectives,
  ],
  destinyMembershipId: '123456',
  getAllMemberships: true,
  itemId: '0987654321',
  itemInstanceId: '0987654321',
  itemReferenceHash: 45674576,
  membershipId: '123456',
  membershipType: 3,
  stackSize: 7,
  transferToVault: true,
  vendorHash: 45674576,
  state: true,
  groupId: '78346592834756',
  correlationId: '456734563673',
  count: 10,
  mode: 5,
  modes: [5, 7],
  page: 2,
  maxtop: 6,
  statid: '234624564',
  returnOriginalProfile: true,
  displayName: 'asdf',
  entityType: 'DestinyInventoryItemDefinition',
  hashIdentifier: 23463456,
  dayend: 'YYYY-MM-DD',
  daystart: 'YYYY-MM-DD',
  groups: ['General', 'Weapons'],
  periodType: 'AllTime',
  milestoneHash: 23546789,
  collectiblePresentationNodeHash: 10293847,
  activityId: '456745673456',
  filter: 0,
  type: 'DestinyInventoryItemDefinition',
  searchTerm: 'something',
};

const testCases: [(..._: any) => any, any[]][] = [
  [getDestinyManifest, [h]],
  [getDestinyEntityDefinition, [h, p]],
  [searchDestinyPlayer, [h, p]],
  [getLinkedProfiles, [h, p]],
  [getProfile, [h, p]],
  [getCharacter, [h, p]],
  [getClanWeeklyRewardState, [h, p]],
  [getItem, [h, p]],
  [getVendors, [h, p]],
  [getVendor, [h, p]],
  [getPublicVendors, [h, p]],
  [getCollectibleNodeDetails, [h, p]],
  [transferItem, [h, b]],
  [pullFromPostmaster, [h, b]],
  [equipItem, [h, b]],
  [equipItems, [h, b]],
  [setItemLockState, [h, b]],
  [setQuestTrackedState, [h, b]],
  [insertSocketPlug, [h, b]],
  [getPostGameCarnageReport, [h, p]],
  [reportOffensivePostGameCarnageReportPlayer, [h, p, b]],
  [getHistoricalStatsDefinition, [h]],
  [getClanLeaderboards, [h, p]],
  [getClanAggregateStats, [h, p]],
  [getLeaderboards, [h, p]],
  [getLeaderboardsForCharacter, [h, p]],
  [searchDestinyEntities, [h, p]],
  [getHistoricalStats, [h, p]],
  [getHistoricalStatsForAccount, [h, p]],
  [getActivityHistory, [h, p]],
  [getUniqueWeaponHistory, [h, p]],
  [getDestinyAggregateActivityStats, [h, p]],
  [getPublicMilestoneContent, [h, p]],
  [getPublicMilestones, [h]],
  [awaInitializeRequest, [h, b]],
  [awaProvideAuthorizationResult, [h, b]],
  [awaGetActionToken, [h, p]],
];

test.each(testCases)('check config builder for %p', async (func, args) => {
  expect(await func(...args)).toMatchSnapshot();
});
