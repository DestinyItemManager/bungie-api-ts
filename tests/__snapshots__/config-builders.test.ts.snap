// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check config builder for [Function awaGetActionToken] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Awa/GetActionToken/456734563673/",
}
`;

exports[`check config builder for [Function awaInitializeRequest] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Awa/Initialize/",
}
`;

exports[`check config builder for [Function awaProvideAuthorizationResult] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Awa/AwaProvideAuthorizationResult/",
}
`;

exports[`check config builder for [Function equipItem] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/EquipItem/",
}
`;

exports[`check config builder for [Function equipItems] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/EquipItems/",
}
`;

exports[`check config builder for [Function getActivityHistory] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "count": 10,
    "mode": 5,
    "page": 2,
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Account/123456/Character/1234658790/Stats/Activities/",
}
`;

exports[`check config builder for [Function getCharacter] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/Character/1234658790/",
}
`;

exports[`check config builder for [Function getClanAggregateStats] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "modes": Array [
      5,
      7,
    ],
  },
  "url": "https://www.bungie.net/Platform/Destiny2/Stats/AggregateClanStats/78346592834756/",
}
`;

exports[`check config builder for [Function getClanLeaderboards] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "maxtop": 6,
    "modes": Array [
      5,
      7,
    ],
    "statid": "234624564",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/Stats/Leaderboards/Clans/78346592834756/",
}
`;

exports[`check config builder for [Function getClanWeeklyRewardState] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Clan/78346592834756/WeeklyRewardState/",
}
`;

exports[`check config builder for [Function getCollectibleNodeDetails] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/Character/1234658790/Collectibles/10293847/",
}
`;

exports[`check config builder for [Function getDestinyAggregateActivityStats] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/3/Account/123456/Character/1234658790/Stats/AggregateActivityStats/",
}
`;

exports[`check config builder for [Function getDestinyEntityDefinition] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Manifest/DestinyInventoryItemDefinition/23463456/",
}
`;

exports[`check config builder for [Function getDestinyManifest] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Manifest/",
}
`;

exports[`check config builder for [Function getHistoricalStats] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "dayend": "YYYY-MM-DD",
    "daystart": "YYYY-MM-DD",
    "groups": "General,Weapons",
    "modes": "5,7",
    "periodType": "AllTime",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Account/123456/Character/1234658790/Stats/",
}
`;

exports[`check config builder for [Function getHistoricalStatsDefinition] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Stats/Definition/",
}
`;

exports[`check config builder for [Function getHistoricalStatsForAccount] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "groups": "General,Weapons",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Account/123456/Stats/",
}
`;

exports[`check config builder for [Function getItem] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/Item/0987654321/",
}
`;

exports[`check config builder for [Function getLeaderboards] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "maxtop": 6,
    "modes": Array [
      5,
      7,
    ],
    "statid": "234624564",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Account/123456/Stats/Leaderboards/",
}
`;

exports[`check config builder for [Function getLeaderboardsForCharacter] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "maxtop": 6,
    "modes": Array [
      5,
      7,
    ],
    "statid": "234624564",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/Stats/Leaderboards/3/123456/1234658790/",
}
`;

exports[`check config builder for [Function getLinkedProfiles] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "getAllMemberships": true,
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/LinkedProfiles/",
}
`;

exports[`check config builder for [Function getPostGameCarnageReport] 1`] = `
Object {
  "method": "GET",
  "url": "https://stats.bungie.net/Platform/Destiny2/Stats/PostGameCarnageReport/456745673456/",
}
`;

exports[`check config builder for [Function getProfile] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/",
}
`;

exports[`check config builder for [Function getPublicMilestoneContent] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Milestones/23546789/Content/",
}
`;

exports[`check config builder for [Function getPublicMilestones] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/Milestones/",
}
`;

exports[`check config builder for [Function getPublicVendors] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/Vendors/",
}
`;

exports[`check config builder for [Function getUniqueWeaponHistory] 1`] = `
Object {
  "method": "GET",
  "url": "https://www.bungie.net/Platform/Destiny2/3/Account/123456/Character/1234658790/Stats/UniqueWeapons/",
}
`;

exports[`check config builder for [Function getVendor] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/Character/1234658790/Vendors/45674576/",
}
`;

exports[`check config builder for [Function getVendors] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "components": "400,402,300,301,304,305,306,307,600,308,310,309",
    "filter": 0,
  },
  "url": "https://www.bungie.net/Platform/Destiny2/3/Profile/123456/Character/1234658790/Vendors/",
}
`;

exports[`check config builder for [Function insertSocketPlug] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/InsertSocketPlug/",
}
`;

exports[`check config builder for [Function pullFromPostmaster] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/PullFromPostmaster/",
}
`;

exports[`check config builder for [Function reportOffensivePostGameCarnageReportPlayer] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://stats.bungie.net/Platform/Destiny2/Stats/PostGameCarnageReport/456745673456/Report/",
}
`;

exports[`check config builder for [Function searchDestinyEntities] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "page": 2,
  },
  "url": "https://www.bungie.net/Platform/Destiny2/Armory/Search/DestinyInventoryItemDefinition/something/",
}
`;

exports[`check config builder for [Function searchDestinyPlayer] 1`] = `
Object {
  "method": "GET",
  "params": Object {
    "returnOriginalProfile": true,
  },
  "url": "https://www.bungie.net/Platform/Destiny2/SearchDestinyPlayer/3/asdf/",
}
`;

exports[`check config builder for [Function setItemLockState] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/SetLockState/",
}
`;

exports[`check config builder for [Function setQuestTrackedState] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/SetTrackedState/",
}
`;

exports[`check config builder for [Function transferItem] 1`] = `
Object {
  "body": Object {},
  "method": "POST",
  "url": "https://www.bungie.net/Platform/Destiny2/Actions/Items/TransferItem/",
}
`;
